src/main/java/com/mraof/minestuck/inventory/AtheneumMenu.java

```diff
@@ -0,0 +1,207 @@
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AtheneumMenu extends AbstractContainerMenu
+{
+	private final int INVENTORY_SIZE = 21;
+	private final int INVENTORY_COLUMNS = 7;
+	private final Player player;
+	private final Container inventory = new SimpleContainer(INVENTORY_SIZE);
+	private List<ItemStack> items  = new ArrayList<>();
+	private int scroll;
+	
+	public AtheneumMenu(int windowId, Inventory playerInventory)
+	{
+		super(MSMenuTypes.ATHENEUM.get(), windowId);
+		this.player = playerInventory.player;
+		addSlots();
+		if(player instanceof ServerPlayer)
+		{
+			updateInventory();
+		}
+	}
+	
+	public void updateScroll(boolean scrollUp)
+	{
+		scroll += scrollUp ? -1 : 1;
+		scroll = Mth.clamp(scroll, 0, items.size()/INVENTORY_COLUMNS);
+		
+		sendPacket();
+	}
+	
+	@Override
+	public boolean stillValid(Player player)
+	{
+		return player == this.player;
+	}
+	
+	@Override
+	public ItemStack quickMoveStack(Player player, int slotIndex)
+	{
+		if(slotIndex >= INVENTORY_SIZE && slotIndex < this.slots.size())
+		{
+			Slot slot = this.slots.get(slotIndex);
+			ItemStack stack = slot.getItem();
+			slot.set(ItemStack.EMPTY);
+			return stack;
+		}
+		if(slotIndex >= 0 && slotIndex < INVENTORY_SIZE)
+		{
+			Slot slot = this.slots.get(slotIndex);
+			ItemStack stack = slot.getItem();
+			if(!stack.isEmpty())
+				for(int i = INVENTORY_SIZE; i < slots.size(); i++)
+					if(!getSlot(i).hasItem())
+					{
+						getSlot(i).set(stack);
+						return stack;
+					}
+		}
+		return ItemStack.EMPTY;
+	}
+	
+	private void addSlots()
+	{
+		
+		for(int i = 0; i < INVENTORY_SIZE; i++)
+				addSlot(new InventorySlot(inventory, i, 26+(i%INVENTORY_COLUMNS)*18, 14+(i/INVENTORY_COLUMNS)*18));
+		
+		for(int i = 0; i < 9; i++)
+			addSlot(new ToolbarSlot(player.getInventory(), i, 8+i*18, 74));
+	}
+	
+	private void updateInventory()
+	{
+		EditData editData = ServerEditHandler.getData(player);
+		if(editData == null)
+			throw new IllegalStateException("Creating an editmode inventory menu, but the player is not in editmode");
+		List<ItemStack> itemList = new ArrayList<>();
+		SburbConnection c = editData.getConnection();
+		
+		List<DeployEntry> atheneumItems = DeployList.getItemList(player.getServer(), c, DeployList.EntryLists.ATHENEUM);
+		atheneumItems.removeIf(deployEntry -> deployEntry.getCurrentCost(c) == null);
+		
+		for(DeployEntry atheneumItem : atheneumItems)
+		{
+			if(!atheneumItem.getItemStack(c, player.level).isEmpty())
+				itemList.add(atheneumItem.getItemStack(c, player.level));
+		}
+		
+		boolean changed = false;
+		if(itemList.size() != this.items.size())
+			changed = true;
+		else for(int i = 0; i < itemList.size(); i++)
+			if(!ItemStack.matches(itemList.get(i), this.items.get(i)))
+			{
+				changed = true;
+				break;
+			}
+		
+		if(changed)
+		{
+			this.items = itemList;
+			sendPacket();
+		}
```

Comment:
Its a little dense to read. Mind adding at least two comments total on whatever code blocks you think could use an explanation the most and/or extract some code blocks into methods of their own?